### PR TITLE
Add Rails 7.2 support

### PR DESCRIPTION
**Pull request by AhmedAliIbrahim**
Adds Rails 7.2 support

### CODE REVIEW

File: awesome_nested_set.gemspec [hunk 1]
```diff
@@ -18,7 +18,7 @@ Gem::Specification.new do |s|
 
   s.required_ruby_version = '>= 2.0.0'
 
-  s.add_runtime_dependency 'activerecord', '>= 4.0.0', '< 7.2'
+  s.add_runtime_dependency 'activerecord', '>= 4.0.0', '< 8.0'
```
Comment (ngrebenshikov) [inline awesome_nested_set.gemspec:21]: Why 8.0 instead of 7.3?

Comment (AhmedAliIbrahim) [reply awesome_nested_set.gemspec:21]: because the next release will be 8.0 as I know

Comment (ngrebenshikov) [reply awesome_nested_set.gemspec:21]: It's good to add the tests configuration for Rails 7.2 like it's done for 7.1 in https://github.com/collectiveidea/awesome_nested_set/commit/034ab03ce97e5f078581cffb70e2385a06415e8a

Comment (AhmedAliIbrahim) [reply awesome_nested_set.gemspec:21]: Ok, will do that

Comment (AhmedAliIbrahim) [reply awesome_nested_set.gemspec:21]: @ngrebenshikov updated!

**Review by ngrebenshikov — APPROVED** [review]
LGTM

**Review by ngrebenshikov — APPROVED** [review]
LGTM

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -50,6 +51,9 @@ jobs:
 
           - ruby: '2.6'
             gemfile: rails_7_1
+
+          - ruby: '3.1'
```
[the remark this answers is not in the document]
Comment (AhmedAliIbrahim) [reply .github/workflows/ci.yml]: I've fixed it already it was a mistake

**Comment by ngrebenshikov** [comment]
@parndt , could you, please, take a look at this PR and give some advises how to move it forward?

**Comment by parndt** [comment]
@AhmedAliIbrahim @ngrebenshikov I fixed the CI issue where it was cancelling jobs, but it seems a lot of them are failing.

File: Appraisals [hunk 3]
```diff
@@ -1,10 +1,17 @@
...
+}.each do |rails, version|
   appraise "rails-#{rails}" do
     gem "rails", "~> #{version}"
   end
 end
 
 appraise "rails-main" do
   gem "rails", github: "rails/rails", branch: "main"
-  gem "rspec-rails", "~> 6.0.2"
+  gem "rspec-rails", "~> 6.1"
```
Comment (parndt) [inline Appraisals:16]:
```suggestion
  gem "rspec-rails", "~> 6.0.2"
```

File: gemfiles/rails_7_0.gemfile [hunk 4]
```diff
@@ -5,7 +5,7 @@ source "https://rubygems.org"
 gem "rails", "~> 7.0.0"
 
 platforms :ruby do
-  gem "sqlite3"
+  em "sqlite3", "~> 1.4"
```
Comment (nickgal) [inline gemfiles/rails_7_0.gemfile:8]: `gem` typo

**Review by parndt — APPROVED** [review]

**Comment by parndt** [comment]
Brilliant, thank you @AhmedAliIbrahim

**Pull request merged**